File: packages/i18nify-js/src/modules/banking/__tests__/getPaymentNetwork.test.ts
import getPaymentNetwork from '../getPaymentNetwork';

describe('getPaymentNetwork', () => {
  it('returns payment networks for INR', () => {
    expect(getPaymentNetwork('INR')).toEqual({
      networks: ['UPI', 'IMPS', 'NEFT', 'RTGS'],
      primary: 'UPI',
    });
  });

  it('is case-insensitive', () => {
    expect(getPaymentNetwork('usd').primary).toBe('ACH');
  });

  it('throws for empty input', () => {
    expect(() => getPaymentNetwork('')).toThrow(
      "Parameter 'currencyCode' is invalid! The received value was: .",
    );
  });

  it('throws for unsupported currencies', () => {
    expect(() => getPaymentNetwork('XXX')).toThrow(
      'Currency code "XXX" is not supported for payment network lookup.',
    );
  });
});

File: packages/i18nify-go/modules/banking/get_payment_network_test.go
package banking

import "testing"

func TestGetPaymentNetwork(t *testing.T) {
	t.Run("returns payment networks for INR", func(t *testing.T) {
		got, err := GetPaymentNetwork("INR")
		if err != nil {
			t.Fatalf("expected no error, got %v", err)
		}
		if got.Primary != "UPI" || len(got.Networks) != 4 {
			t.Fatalf("unexpected result: %+v", got)
		}
	})

	t.Run("lookup is case-insensitive", func(t *testing.T) {
		got, err := GetPaymentNetwork("usd")
		if err != nil {
			t.Fatalf("expected no error, got %v", err)
		}
		if got.Primary != "ACH" {
			t.Fatalf("unexpected primary network: %s", got.Primary)
		}
	})

	t.Run("errors on empty input", func(t *testing.T) {
		if _, err := GetPaymentNetwork(""); err == nil {
			t.Fatal("expected an error for empty input")
		}
	})

	t.Run("errors on unsupported currencies", func(t *testing.T) {
		if _, err := GetPaymentNetwork("XXX"); err == nil {
			t.Fatal("expected an error for unsupported currency")
		}
	})
}
